### DISCUSSION TITLE
Issue with pushing files more than 100Mb

### DISCUSSION THREAD

**Discussion by NUS3BAN** [community/community#61810] [New to GitHub]
Hi Everyone,
 
I am trying to push the files which is more than 100 Mb using git Lfs but still getting error, if any one able to do this please suggest me the steps.
Error:
remote: error: File iService5/iService5.Core.Tests/DatabaseFiles/MetadataDownloadFiltering.db3 is 105.28 MB; this exceeds GitHub's file size limit of 100.00 MB
! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs
Thanks

**Comment by nirav-gajera** [comment]
To push this file to GitHub, you will need to use Git Large File Storage (LFS). LFS is a way to store large files in a separate repository from your main Git repository. This allows you to track large files in your Git history, but they are not actually stored in the main repository.

To use LFS, you will need to install the LFS client on your computer. Once you have installed the client, you can track the file `MetadataDownloadFiltering.db3` with LFS. To do this, run the following command:

```
git lfs track iService5/iService5.Core.Tests/DatabaseFiles/MetadataDownloadFiltering.db3
```

This will tell LFS to track the file `MetadataDownloadFiltering.db3` and store it in a separate repository. Once you have tracked the file, you can push it to GitHub as usual.

Here are the steps to push a large file to GitHub using Git LFS:

1. Install the LFS client on your computer.
2. Track the large file with LFS.
3. Commit the changes to your local repository.
4. Push the changes to GitHub.

Here are the specific commands you need to run:

```
# Install the LFS client
git lfs install

# Track the large file
git lfs track iService5/iService5.Core.Tests/DatabaseFiles/MetadataDownloadFiltering.db3

# Commit the changes
git commit -am "Tracked large file"

# Push the changes
git push
```

Once you have pushed the changes, the file `MetadataDownloadFiltering.db3` will be stored in a separate repository on GitHub. You can then access the file from the GitHub website or from the command line.

**Reply by NUS3BAN** [reply]
nirav-gajera thank you for the reply .I tried all the commands which you suggested still am getting the error.

**Reply by nirav-gajera** [reply]
* Make sure that you have added the `.gitattributes` file to your repository. You can do this by running the following command:

```
git status
```

If the `.gitattributes` file is not listed, then you need to add it to your repository.

* Make sure that the file you are trying to push is actually tracked by Git LFS. You can do this by running the following command:

```
git lfs status
```

If the file is not listed, then you need to track it by running the `git lfs track` command.

* Try restarting your computer and clearing the cache.

* If you are using a proxy server, make sure that you have configured Git LFS to use the proxy server. You can do this by editing the `lfs.conf` file.

If you are still having trouble, you can contact GitHub support for help.

Here is a link to the GitHub support documentation on Git LFS: https://docs.github.com/en/repositories/working-with-files/managing-large-files/configuring-git-large-file-storage

**Reply by NUS3BAN** [reply]
yes i did all the above things but no luck still one file which is more than 100Mb throwing the error even if we track same file with git lfs .

**Reply by nirav-gajera** [reply]
check out this documentation you'll find something : https://docs.github.com/en/repositories/working-with-files/managing-large-files/configuring-git-large-file-storage
